### PR TITLE
Extension serve unit tests

### PR DESCRIPTION
**Pull request by ulgut**
PR for reducing the test code bloat in `app_extension_proxy_test.rb` and `theme_proxy_test.rb`

### CODE REVIEW

File: test/shopify-cli/theme/dev_server/app_extension_proxy_test.rb [hunk 1]
```diff
@@ -158,98 +69,6 @@ def test_pass_replace_templates_from_cookie_to_storefront
...
-          stub_request(:post, "https://dev-theme-server-store.myshopify.com/cart/add?_fd=0&pb=0")
-            .with(
-              headers: default_proxy_headers.merge(
-                "Content-Length" => "272",
-                "Content-Type" => "multipart/form-data; boundary=AaB03x",
-              )
-            )
-            .to_return(status: 200)
-
-          # TODO: fix -- using an theme app extension file causes the test to fail, using a theme file works
```
Comment (jessetuglu) [inline test/shopify-cli/theme/dev_server/app_extension_proxy_test.rb:172]: I moved to `base_proxy_test.rb` but will revisit this

File: test/shopify-cli/theme/dev_server/base_proxy_test.rb [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+              headers: default_proxy_headers,
+            )
+            .to_return(status: 200)
+            .times(2)
+
+          stub_session_id_request
+          request.get("/")
+
+          # Should refresh the session cookie after 1 day
+          Timecop.freeze(DateTime.now + 1) do # rubocop:disable Style/DateTime
```
Comment (karreiro) [inline test/shopify-cli/theme/dev_server/base_proxy_test.rb:46]: We could be RuboCop friendly by changing this line to:
```suggestion
          Timecop.freeze(tomorrow) do
```

Then, we could add the tomorrow method:

```ruby
def tomorrow
  Time.now + 86400
end
```

File: test/shopify-cli/theme/dev_server/base_proxy_test.rb [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+          assert_requested(:head,
+            "https://dev-theme-server-store.myshopify.com/?_fd=0&pb=0&preview_theme_id=123456789",
+            times: 2)
+        end
+
+        def test_update_session_cookie_when_returned_from_backend
+          stub_session_id_request
+          new_secure_session_id = "#{SECURE_SESSION_ID}2"
+
+          # POST response returning a new session cookie (Set-Cookie)
```
Comment (karreiro) [inline test/shopify-cli/theme/dev_server/base_proxy_test.rb:59]: The code is so clear that it makes me find these comments a bit redundant (but feel free to keep them if you prefer) :)

**Review by karreiro — APPROVED** [review]
Great stuff, @jessetuglu! LGTM 🚀  I've only two minor comments :)

File: test/shopify-cli/theme/dev_server/base_proxy_test.rb (hunk 3 above)
Comment (jessetuglu) [reply test/shopify-cli/theme/dev_server/base_proxy_test.rb:59]: Thanks! I'll leave it as is, but will keep these unresolved in case I revisit later? 👍

**Pull request merged**